### DISCUSSION TITLE
Tam Sayıdan Romana

### DISCUSSION THREAD

**Discussion by adasue** [sonsuzus/sonsuzus#370] [Soru Cevap] [answered]
_Roman rakamları 7 farklı sembol ile gösterilir I, V, X, L, C, D ve M._
**_Sembol Değer_**
I 1
V 5
X 10
L 50
C 100
D 500
M 1000

Örneğin 2 roman rakamı olarak II şeklinde birbirine eklenmiş iki bir yazılır. 12 ise XII diye yazılır, basitçe X + II. 27, XXVII şeklinde yazılır diğer şekilde XX + V + II.
Roman rakamları genellikle en genişten en küçüğe doğru soldan sağa yazılır. Fakat 4 numarası IIII şeklinde değil. Onun yerine numara 4, IV şeklinde yazılır. Çünkü beşten önceki biri kendisinden çıkararak dört elde ediyoruz. Benzer prensip dokuz için de geçerlidir, IX şeklinde yazılan. Çıkarmanın kullanıldığı 6 örnek vardır:
I, V(5) ve X(10) öncesine yerleştirilerek 4 ve 9 yapılabilir.
X, L(50) ve C(100) öncesine yerleştirilerek 40 ve 90 yapılabilir.
C, D(500) ve M(1000) öncesine yerleştirilerek 400 ve 900 yapılabilir.

Verilen tam sayıyı, roman rakamına dönüştürün.
**Örnek 1:**
Giriş: num = 3
Çıktı: “III1
Açıklama: 3, 3 tane 1 olarak gösterilir.

**Örnek 2:**
Giriş: num = 58
Çıktı: “LVIII”
Açıklama: L = 50, V = 5, III = 3.

**Örnek 3:**
Giriş: num = 1994
Çıktı: “MCMXCIV”
Açıklama: M = 1000, CM = 900, XC = 90 ve IV = 4.
_Kısıtlamalar:_
1<= num <= 3999

Kaynakça: https://leetcode.com/problems/integer-to-roman/

**Comment by nilbuyukkinaci** [comment, marked as answer]
int num;
	
string birler[] = {"", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX"};
string onlar[] = {"", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC"};
string yuzler[] = {"", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM"};
string binler[]={"", "M", "MM", "MMM"};
	
cin >> num;
	
string romen = binler[num / 1000] + yuzler[(num % 1000) / 100] + onlar[(num % 100) / 10] + birler[num % 10];
	
cout << romen;
